numba/cuda/cudadrv/driver.py

```diff
@@ -2644,6 +2655,85 @@ def complete(self):
         """
 
 
+class MVCLinker(Linker):
+    """
+    Linker supporting Minor Version Compatibility, backed by the cubinlinker
+    package.
+    """
+    def __init__(self, max_registers=None, lineinfo=False, cc=None):
+        if cc is None:
+            raise RuntimeError("MVCLinker requires CC to be specified")
+
+        sm_ver = f"{cc[0] * 10 + cc[1]}"
+        ptx_compile_opts = ['--gpu-name', f'sm_{sm_ver}', '-c']
+        if max_registers:
+            arg = f"--maxrregcount={max_registers}"
+            ptx_compile_opts.append(arg)
+        if lineinfo:
+            ptx_compile_opts.append('--generate-line-info')
+        self.ptx_compile_options = tuple(ptx_compile_opts)
+
+        arch = f"--arch=sm_{sm_ver}"
+        self._linker = CubinLinker(arch)
+
+    @property
+    def info_log(self):
+        return self._linker.info_log
+
+    @property
+    def error_log(self):
+        return self._linker.error_log
+
+    def add_ptx(self, ptx, name='<cudapy-ptx>'):
+        compile_result = compile_ptx(ptx.decode(), self.ptx_compile_options)
+        try:
+            self._linker.add_cubin(compile_result.compiled_program, name)
+        except CubinLinkerError as e:
+            raise LinkerError from e
+
+    def add_file(self, path, kind):
+        try:
+            with open(path, 'rb') as f:
```

Comment:
It might micro-optimize performance for a specific error case, but it seems to make the logic slightly more convoluted because we take different actions later on based on the `kind` as well.

Reply:
OK, let's leave as-is for code clarity.